src/clients/meta/MetaClient.cpp

```diff
@@ -2353,16 +2412,73 @@ std::vector<cpp2::RoleItem> MetaClient::getRolesByUserFromCache(const std::strin
   return iter->second;
 }
 
-bool MetaClient::authCheckFromCache(const std::string& account, const std::string& password) {
+Status MetaClient::authCheckFromCache(const std::string& account, const std::string& password) {
+  // Check meta service status
   if (!ready_) {
-    return false;
+    return Status::Error("Meta Service not ready");
   }
+
   const ThreadLocalInfo& threadLocalInfo = getThreadLocalInfo();
+  // Check user existence
   auto iter = threadLocalInfo.userPasswordMap_.find(account);
   if (iter == threadLocalInfo.userPasswordMap_.end()) {
-    return false;
+    return Status::Error("User not exist");
+  }
+
+  folly::RWSpinLock::WriteHolder holder(localCacheLock_);
+
+  auto& lockedSince = userLoginLockTime_[account];
```

Comment:
what if the account does not exist in the map?

Reply:
user existence will be checked in line 2422

```
> ./nebula-console -addr 192.168.8.6 -port 29562 -u ro -p 1
2021/12/29 22:48:57 Fail to create a new session from connection pool, fail to authenticate, error: User not exist
panic: Fail to create a new session from connection pool, fail to authenticate, error: User not exist
```